scripts/bo_benchmarks/demonstrations/branin_campaign_demonstration.py

```diff
@@ -0,0 +1,419 @@
+from ax.modelbridge.registry import Models
+import matplotlib.pyplot as plt
+import torch
+from gpcheck.models import GPModel, GPConfig
+from gpcheck.metrics import loo_pseudo_likelihood, importance_concentration
+from sklearn.metrics import r2_score
+from scipy.stats import kendalltau
+
+obj1_name = "branin"
+
+
+def branin(x1, x2):
+    y = float(
+        (x2 - 5.1 / (4 * np.pi**2) * x1**2 + 5.0 / np.pi * x1 - 6.0) ** 2
+        + 10 * (1 - 1.0 / (8 * np.pi)) * np.cos(x1)
+        + 10
+    )
+    return y
+
+
+def interval_score(y_true, lower, upper, alpha=0.05):
+    """
+    Calculate interval score for uncertainty quantification evaluation.
+    
+    For 95% confidence intervals (alpha=0.05), the formula is:
+    f_interval = (upper - lower) + (2/alpha) * max(0, lower - y) + (2/alpha) * max(0, y - upper)
+    
+    Args:
+        y_true: True values
+        lower: Lower confidence bounds
+        upper: Upper confidence bounds 
+        alpha: Confidence level (0.05 for 95% CI)
+    
+    Returns:
+        Array of interval scores (lower is better)
+    """
+    width = upper - lower
+    below_penalty = np.maximum(0, lower - y_true)
+    above_penalty = np.maximum(0, y_true - upper)
+    return width + (2 / alpha) * below_penalty + (2 / alpha) * above_penalty
+
+
+def normal_95ci(mean, std):
+    """Convert mean and standard deviation to 95% confidence intervals."""
+    lower = mean - 1.96 * std
+    upper = mean + 1.96 * std
+    return lower, upper
+
+
+def calculate_ax_cross_validation_metrics(ax_client_cv, trial_index, x1, x2, result):
+    """
+    Calculate metrics using a separate Ax client dedicated to cross-validation.
+    This client skips Sobol initialization and uses BOTORCH_MODULAR immediately.
+    """
+    try:
+        # Skip if not enough data for cross-validation
+        if trial_index < 3:
+            return None, None
+        
+        # Attach the current trial to the CV client (mirroring main client data)
+        params = {"x1": x1, "x2": x2}
+        _, trial_index_cv = ax_client_cv.attach_trial(parameters=params)
+        ax_client_cv.complete_trial(trial_index=trial_index_cv, raw_data=result)
+        
+        # Force the CV client to generate a trial to trigger model creation if needed
+        # We don't use this trial, it's just to ensure the model bridge is created
+        if len(ax_client_cv.experiment.trials) >= 3 and ax_client_cv.generation_strategy.model is None:
+            try:
+                _, dummy_trial_idx = ax_client_cv.get_next_trial()
+                # Abandon the dummy trial - we just needed to trigger model creation
+                ax_client_cv.abandon_trial(trial_index=dummy_trial_idx)
+            except Exception:
+                pass  # Ignore errors in dummy trial generation
+        
+        # Check if CV client has a Bayesian model
+        # The client should switch to BOTORCH_MODULAR once it has enough data
+        if len(ax_client_cv.experiment.trials) < 3:
```

Comment:
seems like too much logic / overengineering here. Can't we make this a bit simpler, i.e., assume that Ax is doing what it should in having an actual model, likewise especially since we're skipping directly to a model based approach after having attached some data already? Likewise, it seems that calculate_iteration_metrics already has a bypass for trial_index < 3